src/ansys/fluent/core/workflow.py

```diff
@@ -1109,18 +1155,14 @@ def __dir__(self):
         """Override the behaviour of dir to include attributes in WorkflowWrapper and
         the underlying workflow."""
         arg_list = [camel_to_snake_case(arg) for arg in dir(self._workflow)]
-        return list(
-            dict.fromkeys(
-                sorted(
-                    set(
-                        list(self.__dict__.keys())
-                        + dir(type(self))
-                        + arg_list
-                        + self.child_task_python_names()
-                    )
-                )
-            )
+        dir_list = set(
+            list(self.__dict__.keys())
+            + dir(type(self))
+            + arg_list
+            + self.child_task_python_names()
         )
+        dir_list = dir_list - self._unwanted_attrs
+        return sorted(dir_list)
```

Comment:
```suggestion
        dir_set = set(
            list(self.__dict__.)
            + dir(type(self))
            + arg_list
            + self.child_task_python_names()
        )
        dir_set = dir_set - self._unwanted_attrs
        return sorted(filter(None, dir_set))
```
A few changes here again - the last line is the key one!

Reply:
N.b. python names of dynamic tasks are None so they had to be filtered out here. Otherwise `-` raises!